feat: 로그인페이지 UI 구현

soo96 opened the pull request:
## 요약

- 로그인페이지 UI 구현

Close #10 

## 변경 사항

- [x]  로그인페이지 UI 구현
- [x]  로그인페이지에서 소셜 인증 로그인 요청 기능 구현 (카카오, 네이버, 구글)
- [x] OAuth 인증 서버로부터 인증코드 요청
- [x] API 서버에 로그인 요청

## 체크리스트

- [x] 이해하기 어려운 코드에는 주석을 추가했습니다  
- [x] 관련 문서를 수정했습니다  
- [x] 새로운 경고가 발생하지 않습니다  
- [x] 수정한 기능 또는 추가한 기능에 대해 테스트를 작성했습니다  
- [x] 의존성 있는 변경 사항이 병합 및 배포되었습니다

## 관련 이슈

- #10 
- #23 

## 리뷰 반영 사항

<!-- PR리뷰 반형 사항 -->

<details>
<summary><strong>선택 항목 펼치기</strong></summary>

## 스크린샷

<!-- 시각적인 변경 사항이 있다면 스크린샷이나 GIF를 포함해 주세요. 리뷰어가 쉽게 이해할 수 있습니다. -->

## 테스트 방법

<!-- PR에서 수행한 변경 사항을 어떻게 테스트할 수 있는지 설명해 주세요. 리뷰어가 직접 확인할 수 있도록 돕습니다. -->

## 리뷰어 참고사항

<!-- 리뷰어에게 특별히 알려야 할 사항이나 고려할 점이 있다면 여기에 작성해 주세요. -->

</details>

soo96 committed:
chore: 소셜로그인 아이콘 이미지 추가

soo96 committed:
feat: 소셜 로그인 UI 및 인증코드 요청 기능 추가

soo96 committed:
feat: 인증 서버 리다이렉트 처리 및 백엔드 로그인 API 연동

soo96 committed:
chore: dev브랜치 최신 사항 반영

soo96 committed:
test: App.spec.tsx 로그인 테스트코드 수정

soo96 committed:
chore: .gitignore 파일 환경변수 설정 수정

leesm0218 reviewed (approved):
style 들은 layout 도 없는 페이지고 코드도 몇줄 안 되서 괜찮은 것 같습니다. 수고하셨습니다.

smileboy0014 commented on src/components/Header/index.tsx:
```diff
@@ -0,0 +1,9 @@
+};
+
+const Header = ({ headerTitle }: HeaderProps) => {
+  return <h1 className="text-2xl font-bold mb-8">{headerTitle}</h1>;
```
```suggestion
const Header = ({ headerTitle }: HeaderProps) => {
  return <h1 className="HEADER_STYLE.titleText">{headerTitle}</h1>;
```
이런식으로 costants/styles.ts 밑에 빼야하지 않을까요?

soo96 replied:
수정했습니다!🙂

smileboy0014 commented on src/components/Header/index.tsx:
```diff
@@ -0,0 +1,9 @@
+type HeaderProps = {
+  headerTitle: string;
+};
```
props 정보도 타입이니
types/login.ts 이쪽으로 뺴야하지 않을까?

soo96 replied:
> props 정보도 타입이니
types/login.ts 이쪽으로 뺴야하지 않을까?

@smileboy0014 
props 타입은 해당 컴포넌트에서만 사용한다고 생각해서 해당 컴포넌트 위에 두었습니다. 
어떤 타입이든 타입 선언은 무조건 `/types`밑으로 빼도록 할까요?

smileboy0014 replied:
아직 그런 정책을 저희가 정하지 않아서 그것도 정해야 할 것 같습니다!
만약 정말 그 컴포넌트에서만 쓰인다고 하면 그냥 그 컴포넌트 위에 선언해서 쓰고, 조금이라도 공통으로 쓰일 것 같다고 하면 뺄까요?

leesm0218 replied:
style 의 처리 방침과 같게 하면 될 것 같습니다.

soo96 replied:
> 아직 그런 정책을 저희가 정하지 않아서 그것도 정해야 할 것 같습니다!
만약 정말 그 컴포넌트에서만 쓰인다고 하면 그냥 그 컴포넌트 위에 선언해서 쓰고, 조금이라도 공통으로 쓰일 것 같다고 하면 뺄까요?

조금이라도 공통으로 처리될 수 있는 가능성을 판단한는게 애매한 것 같네요 😢

smileboy0014 commented on src/components/OauthRedirectHandler/index.tsx:
```diff
@@ -0,0 +1,61 @@
+              'Content-Type': 'application/json',
+            },
+          }
+        );
```
저희 api 정보도 api/ 이쪽으로 빠지기로 했으니
api/login.ts 이쪽으로 뺴야하지 않을까요?

soo96 replied:
수정했습니다!🙂

smileboy0014 commented on src/components/OauthRedirectHandler/index.tsx:
```diff
@@ -0,0 +1,61 @@
+  );
+};
+
+export default OauthRedirectHandler;
```
저희 export를 
```suggestion
export const OauthRedirectHandler = () ...
```
이렇게 하기로 얘기한거 아닐까요?
default를 하려면 이렇게 밑에서 따로 해주는게 맞는데, 그럼 컨벤션을 모두 밑에서 export 하는걸로 하는게 맞을 것 같아요!

soo96 replied:
> default를 하려면 이렇게 밑에서 따로 해주는게 맞는데, 그럼 컨벤션을 모두 밑에서 export 하는걸로 하는게 맞을 것 같아요!

`export`를 하게되면 `import`할 때 

- `import OauthRedirectHandler from './components/OauthRedirectHandler/OauthRedirectHandler.tsx';` 

이런식으로 중복되는 경우가 생겨 `export default`를 사용했습니다.

어떻게 통일하는게 좋을까요? @leesm0218 @smileboy0014

smileboy0014 replied:
만약 default를 쓸거면, 저는 그냥 밑에서 다 export 해주는걸로 통일하는게 맞는 것 같습니다!

leesm0218 replied:
index.tsx가 배럴 역할을 하게 하면 될 것 같습니다.

smileboy0014 replied:
배럴 역할을 하게 하면 된다는게 어떤 식인지 설명을 드려도 괜찮을까요?

leesm0218 replied:
지금 현수님이 걱정하시는게, 저걸 `index`가 아닌 `OauthRedirectHandler.tsx` 파일로 만들면

```js
import OauthRedirectHandler from './components/OauthRedirectHandler/OauthRedirectHandler.tsx
```

이렇게 길게 적어야 된다고 그러시는것 같은데,

`OauthRedirectHandler.tsx` 로 빼도 `index` 에서 `export` 해주면 똑같이 

```js
import OauthRedirectHandler from './components/OauthRedirectHandler/
```

처럼 뒤에 폴더명까지만 작성해서 불러올 수 있을겁니다.

smileboy0014 commented on src/components/OauthRedirectHandler/index.tsx:
```diff
@@ -0,0 +1,61 @@
+          navigate('/signup');
+        } else {
+          // TODO: accessToken 저장 로직 수정
+          localStorage.setItem('accessToken', accessToken);
```
저희 accessToken을 계속 localStorage에 저장하실건가요??
임시로 그렇게 놔두신 걸까요?

soo96 replied:
일단 임시로 작성하였습니다.
토큰 보관 방식에 대해 추가 논의가 필요할 것 같습니다.

soo96 replied:
이슈 생성해 두었습니다.
- #22

smileboy0014 commented on src/components/OauthRedirectHandler/index.tsx:
```diff
@@ -0,0 +1,61 @@
+            : { provider, authorizationCode: code };
+
+        const response = await axios.post(
+          `${import.meta.env.VITE_API_BASE_URL}/auth/login`,
```
API_BASE_URL 의 경우, 다른 곳에서도 모두 쓰이는 값이니
```js
// constats/api.ts
export const API_BASE_URL = import.meta.env.VITE_API_BASE_URL;
```
이런식으로 해주시면 다른 사람들도 편하게 사용할 수 있지 않을까요?

soo96 replied:
@smileboy0014 
경태님 말씀대로 API 호출부마다 사용되는 값이니 상수로 따로 빼는 방식이 더 좋을 것 같습니다.😀

혹시 `constats/api.ts` 이 파일에 다른 상수도 추가로 들어갈 내용이 있을까요?
`API_BASE_URL` 이 상수 하나만 들어가도 파일로 따로 관리하는것이 좋을까요?

smileboy0014 replied:
저는 지금
```js
export const API_PREPIX = '/api/v1';
export const API_STALE_TIME = 1000 * 10;

```
이런 정보만  constats/api.ts 여기에 빼두었습니다.
사실 현수님께서 빼신 정보는 서버에 통신하기 위한 proxy 정보이기 때문에

```js
//vite.config.ts
import { defineConfig, loadEnv } from 'vite';
import path from 'path';
import react from '@vitejs/plugin-react';

export default defineConfig(({ mode }) => {
  const env = loadEnv(mode, process.cwd());

  return {
    plugins: [react()],
    resolve: {
      alias: {
        '@': path.resolve(__dirname, 'src'),
      },
    },
    server: {
      proxy: {
        '/api/v1/': {
          target: env.VITE_API_BASE_URL,
          changeOrigin: true,
          rewrite: (path) => path.replace(/^\/api/, '/api'),
        },
      },
    },
  };
});

```
이런식으로 여기서 설정 정보를 넣어두는게 낫지 않을까요?

smileboy0014 commented on src/pages/Auth/Login.tsx:
```diff
@@ -0,0 +1,94 @@
+
+export const Login = () => {
+  return (
+    <div className="flex flex-col items-center justify-center h-screen">
+      <Header headerTitle="로그인" />
+      <p className="text-gray-500 mb-6">SNS 계정으로 로그인하기</p>
```
이것도
constants/styles.ts 로 빠져야 하지 않을까요?

soo96 replied:
수정했습니다!🙂

smileboy0014 commented on src/pages/Auth/Login.tsx:
KakaoLoginButton, NaverLoginButton, GoogleLoginButton 컴포넌트를 잘 분리되어 있는데, 컴포넌트들을 
/components/Button 밑으로 안 옮긴 이유가 있을까요?

soo96 replied:
> KakaoLoginButton, NaverLoginButton, GoogleLoginButton 컴포넌트를 잘 분리되어 있는데, 컴포넌트들을
/components/Button 밑으로 안 옮긴 이유가 있을까요?

@smileboy0014 
해당 컴포넌트들은 로그인 페이지에서만 사용되는 버튼으로 다른 곳에서 재사용할 경우가 없다고 판단하여 한 파일에 두었습니다.
component폴더 밑에 각각 파일로 관리하는것이 좋을까요?

smileboy0014 replied:
재사용성이 있는 컴포넌트만 컴포넌트로 만약에 뺄거면
```
components/
  common/              ← 모든 도메인에서 쓰는 진짜 공용 컴포넌트
    Button.tsx
    Spinner.tsx
    Modal.tsx

  recordings/          ← 특정 도메인/페이지 전용
    FilterButton.tsx
    SortButton.tsx
    RecordingItem.tsx

  profile/
    ProfileCard.tsx
    EditProfileForm.tsx

...
```

이런 방향으로 나가거나
Home/Recording/ 밑에
```
index.tsx
FilterButton.tsx
SortButton.tsx
RecordingItem.tsx
```
이런식으로 가는건 어떨까요?

soo96 replied:
```
components/
  common/              ← 모든 도메인에서 쓰는 진짜 공용 컴포넌트
    Button.tsx
    Spinner.tsx
    Modal.tsx

  recordings/          ← 특정 도메인/페이지 전용
    FilterButton.tsx
    SortButton.tsx
    RecordingItem.tsx

  profile/
    ProfileCard.tsx
    EditProfileForm.tsx

...
```
저는 첫번째로 말씀해주신 구조가 괜찮은 것 같습니다. 🙂
지금 `components` 하위에 만들어지 폴더들 기준이 애매한 것 같아서 차라리 말씀해 주신 내용처럼 구분하는 방식이 더 명확한 것 같네요.
- `commom` : 공용 컴포넌트
- 나머지 : 특정 도메인/페이지 전용

@leesm0218 어떻게 생각하시나요?

leesm0218 replied:
그럼 저번에 말씀하셔서 적용된 conponents/button, checkbox 등의 구조는 사실상 페기라고 봐도 될까요?

작업 구분 없이 components 아래에 바로 button폴더나 checkbox 폴더들이 붙던 구조요.

soo96 committed:
refactor: 소셜로그인 버튼 컴포넌트들 개별파일로 분리

soo96 committed:
refactor: 테일윈드 스타일 클래스 상수로 분리

soo96 committed:
refactor: components, pages 폴더 구조 변경

- components/login 하위로 로그인 관련 컴포넌트 이동
- components/common 하위로 공통 컴포넌트 이동
- login.tsx, signup.tsx 컴포넌트 pages 하위로 위치 변경

soo96 committed:
chore: vite.config.ts 서버 프록시 설정

soo96 committed:
refactor: 로그인 요청 로직을 useLoginMutation 커스텀 훅으로 분리

soo96 committed:
feat: OauthRedirect 페이지에 에러 바운더리 추가

soo96 commented:
@leesm0218 @smileboy0014 
리뷰해주신 내용과 회의내용 반영하여 추가작업 진행했습니다.
확인 부탁드립니다.🙏🏻

smileboy0014 commented:
> @leesm0218 @smileboy0014 리뷰해주신 내용과 회의내용 반영하여 추가작업 진행했습니다. 확인 부탁드립니다.🙏🏻

전체적으로 저희가 정해둔 컨벤션이 잘 지켜진 것 같아서 좋습니다!
Errorboudary같은 경우 로그인 경우가 아니더라도, 저희 메인 페이지에서 에러가 났을 경우 공통 에러 페이지 처리를 해두는것도 좋을 것 같습니다!
고생하셨습니다 🙏

soo96 commented:
> `Errorboudary`같은 경우 로그인 경우가 아니더라도, 저희 메인 페이지에서 에러가 났을 경우 공통 에러 페이지 처리를 해두는것도 좋을 것 같습니다!

말씀 감사합니다! 😊
로그인 과정에서 발생하는 에러는 다른 페이지와 처리가 다를 것 같아 이번에는 로그인 부분에만 한정해 에러 바운더리를 적용했습니다.
추후 다른 페이지에도 필요에 따라 확장하거나, `Global Error Boundary` 형태로 적용해보면 좋을 것 같습니다! 🙏🏻

soo96 commented:
@smileboy0014 @leesm0218 
`ErrorBoundary` 적용에 대한 이슈 생성해 두었습니다. ✅
- #23
